### DISCUSSION TITLE
Looking for advice on how to replace an existing video with a new one and re-generate the sprite, etc

### DISCUSSION THREAD

**Discussion by viv1** [mediacms-io/mediacms#745] [Q&A]
**Describe the issue**
Looking for advice on how to replace an existing video with a new one, and then re trigger the encoding, sprite creation, hls files creation, etc  

I have added a custom step during media upload. This step basically happens right before encoding (Inside "Media.media_init" method). Any new video uploaded goes through this custom step which modifies the uploaded video, and then the encoding happens on this new video. This all works well. 

However, I also have other videos which were uploaded before I had added my custom step. I want to basically regenerate everything (i.e. encoding, sprites, hls files, etc) for these older videos. 

Looking for advice on how I can achieve this. One thing I tried was modifying (through a custom script) the original file under the "media_files" and then calling `media_object.media_init()` for that media file. It does not redo the encoding, nor does it create new sprite files.
 
Thank you.

**Comment by viv1** [comment]
Alright, so `media_object.media_init()` actually does work in my case. I thought it did not work because the actually encoding and other things happen as a background task, which apparently had failed in my case due to other reasons. Sorting that out solved the issue for me.

**Reply by klausmcm** [reply]
You were able to manually trigger the transcoding jobs? Do you have any more details on how to do that?

**Comment by sinaqahremani** [comment]
Is there any updates on the solution?
